docs/REST API/rest-api.md

```diff
@@ -8696,6 +8708,7 @@ Get the results of a software installation.
    "host_display_name": "Marko's MacBook Pro",
    "status": "failed",
    "output": "Installing software...\nError: The operation can’t be completed because the item “Falcon” is in use.",
```

Comment:
### Dev note

For automatic apps, if Fleet can't extract a version from the app to double check if an older version is installed, then Fleet will mark the install as failed with this copy: Failed to verify: The install script finished with exit code 0 but Fleet couldn't extract the version from FalconSensor-6.44.pkg.

For manual apps, if Fleet can't extract a version from the app to double check if the exact version is installed, then Fleet will mark the install as failed with the same copy.

This is also shown in the UI [here](https://www.figma.com/design/4pfUOYy7IyMIrjMH2fuCdU/%2319551-Automatically-install-software-and-scope-software-with-labels?node-id=5503-10100&t=piN5Gu8vzYRvESPT-1).

Reply:
### Dev note

For automatic apps, if Fleet finds that the software is missing or an older version is installed after reinstalling (retrying) once, then Fleet will mark the software as failed with this copy: 

"Failed to verify: The install script finished with exit code 0 but the software is now missing. Fleet already tried to reinstall once."

This is shown in the UI [here](https://www.figma.com/design/4pfUOYy7IyMIrjMH2fuCdU/%2319551-Automatically-install-software-and-scope-software-with-labels?node-id=5503-10126&t=piN5Gu8vzYRvESPT-1).

Reply:
Hey @lucasmrod and @RachelElysia heads up, here's the new `"detail"` field to expose the new types of "Failed" to the IT admin.

Note that we also have a similar `"detail"` field in the `profiles` list in the `GET /hosts/:id` endpoint.

cc @gillespi314

Reply:
^ @jacobshandling updates for your work on the host details page

Reply:
Hey @lucasmrod @jacobshandling heads up that @roperzh doesn't have concerns w/ erroring if we can't extract the version.

The plan is to move forward with adding this erorr:
![Screenshot 2024-08-09 at 4 26 52 PM](https://github.com/user-attachments/assets/fc3a8e23-f8da-4e40-84a7-3bb16a23ba02)

Reply:
@noahtalerman Do we want this whole error string (in red) to be on the API? (this change is missing from this PR).

Reply:
Was just looking into this as well @lucasmrod  – [had a similar conversation around deleting labels,](https://github.com/fleetdm/fleet/pull/20872#discussion_r1712369117) my guess is we want to include the whole string based on that.

Reply:
Oh, but the link component will have to be added by the UI – would the API message then be `"Couldn't add. Fleet couldn't read the version from <sw_package_name>."`?

Reply:
No worries. Backend has access to the filename ("<sw_package_name>").

Reply:
Yeah. Last part ("Learn more") should be added by UI AFAICS.

Reply:
> Do we want this whole error string (in red) to be on the API?

@lucasmrod yep. Thanks for catching that it's not in the PR. I added it here: https://github.com/fleetdm/fleet/pull/20872/files#r1714469004